File: java/playwright/src/test/java/dev/selenium/request/WaitTest.java
package dev.selenium.request;

import com.microsoft.playwright.Request;
import dev.selenium.TestBase;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class WaitTest extends TestBase {

  @Test
  public void waitAndGetResult() {
    Request request =
        page.waitForRequest(
            r -> "image".equals(r.resourceType()), () -> {
                    page.navigate("https://selenium.dev");
                });

    Assertions.assertTrue(request.url().contains("sponsors"));
  }
}

File: dotnet/Playwright/Request/WaitTest.cs
using Microsoft.Playwright;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Playwright.Request;

[TestClass]
public class WaitTest : TestBase
{
    [TestMethod]
    public async Task WaitAndGetResult()
    {
        var request = await Page!.RunAndWaitForRequestAsync(
            async () => await Page.GotoAsync("https://selenium.dev"),
            r => r.ResourceType == "image"
        );

        Assert.IsTrue(request.Url.Contains("sponsors"));
    }
}